libarchive/adapters/archive_read.py

```diff
@@ -267,14 +275,15 @@ def _set_read_context(archive_res, format_code=None, filter_code=None):
             archive_read_support_filter_all(archive_res)
 
 @contextlib.contextmanager
-def _enumerator(opener, entry_cls, format_code=None, filter_code=None):
+def _enumerator(opener, entry_cls, passphrase=None, format_code=None, filter_code=None):
     """Return an archive enumerator from a user-defined source, using a user-
     defined entry type.
     """
 
     archive_res = _archive_read_new()
 
     try:
+        r = _archive_read_add_passphrase(archive_res, passphrase)
```

Comment:
Only call this if not `None`.

Reply:
At first, I write a condition to check if None but after doing some tests, providing None to the API call does not raise an error.

Reply:
Couple of comments:

- By saying "only call this if not "None", do you really mean `if passphrase:`?
- 'r' is a success/failure code, but it isn't checked for failure.  But I'm not sure how a failure would be handled here.  
- The call below this (which was not modified) never returns a value, so this 'r' should be omitted:
    `r = _set_read_context(archive_res, format_code, filter_code)`